chromium_src/chrome/browser/sync/bookmark_sync_service_factory.cc

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/sync/bookmark_sync_service_factory.h"
+#include "build/build_config.h"
+
+#if !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/ui/bookmark/bookmark_prefs_service_factory.h"
+
+// Adding BookmarkPrefsServiceFactory as dependency becasue kShowBookmarBar and
+// kAlwaysShowBookmarkBarOnNTP manage bookmar bar state togher and need to
+// register both prefs same time.
+#define DependsOn                                        \
+  DependsOn(BookmarkPrefsServiceFactory::GetInstance()); \
+  DependsOn
+#endif
+#include "src/chrome/browser/sync/bookmark_sync_service_factory.cc"
```

Comment:
Pls add `#undef` to all `chromium_src/**/*.cc` files.

We have them in both `.h` and `.cc` to extend these files easily at any point. Fighting with still active `#defines` is not what we want.

Reply:
@bridiver asked not to add `#undef` to .cc files earlier

Reply:
@bridiver I strongly suggest to not remove `#undef` in .cc files, otherwise this will be a mess down the road. We have `#undefs` for everything in `chromium_src/**/*.cc` files. I don't think it's a good idea to suddenly drop this practice.

Reply:
done